src/rpc.rs

```diff
@@ -20,140 +21,100 @@ fn is_secure(url: &str) -> bool {
 	re.is_match(url)
 }
 
-pub async fn get_blockhash(url: &str, block: u64) -> Result<String, String> {
+pub async fn get_blockhash(url: &str, block: u64) -> Result<String> {
 	let payload = format!(
 		r#"{{"id": 1, "jsonrpc": "2.0", "method": "chain_getBlockHash", "params": [{}]}}"#,
 		block
 	);
 
-	match hyper::Request::builder()
+	let req = hyper::Request::builder()
 		.method(hyper::Method::POST)
 		.uri(url)
 		.header("Content-Type", "application/json")
 		.body(hyper::Body::from(payload))
-	{
-		Ok(req) => {
-			let resp = if is_secure(url) {
-				let https = HttpsConnector::new();
-				let client = hyper::Client::builder().build::<_, hyper::Body>(https);
-				match client.request(req).await {
-					Ok(resp) => Some(resp),
-					Err(_) => None,
-				}
-			} else {
-				let client = hyper::Client::new();
-				match client.request(req).await {
-					Ok(resp) => Some(resp),
-					Err(_) => None,
-				}
-			};
+		.map_err(|e| anyhow::anyhow!("{}", e))?;
```

Comment:
I think we don't need to wrap errors here, instead of 

```
let req = hyper::Request::builder()
		.method(hyper::Method::POST)
		.uri(url)
		.header("Content-Type", "application/json")
		.body(hyper::Body::from(payload))
		.map_err(|e| anyhow::anyhow!("{}", e))?;
```

we should be able to write  just:

```
let req = hyper::Request::builder()
		.method(hyper::Method::POST)
		.uri(url)
		.header("Content-Type", "application/json")
		.body(hyper::Body::from(payload))?;
```

But since we had some context in the error previously, if we import Context trait from anyhow with:

```use anyhow::{Context, Result};```

we should be able to attach context to error also:

```
	let req = hyper::Request::builder()
		.method(hyper::Method::POST)
		.uri(url)
		.header("Content-Type", "application/json")
		.body(hyper::Body::from(payload))
		.context("Failed to build HTTP POST request object")?;
```

If you agree, this should be done on all places instead of creating error with anyhow! macro.

Reply:
Done